torch/fx/experimental/proxy_tensor.py

```diff
@@ -1371,6 +1372,38 @@ def __torch_function__(
             return node
             # Don't actually run the function! We just want to trace the calls
             # into a graph. We don't actualy want to change global autograd state.
+        if func in [torch._C._functorch._add_batch_dim, torch._C._functorch._remove_batch_dim]:
```

Comment:
this is a very high level take but i'm surprised that so many changes were needed. The naive way I would have implemented this is:

(1) find the location that each of these ~5 APIs is implemented and add a torch_function handler to each (a few might already have torch_function handlers)

(2) Take the existing `TorchFunctionMode` we use in pre-dispatch tracing and tweak it to add these ops to its allow-list of what to proxy